EIPS/eip-7932.md

```diff
@@ -41,132 +41,56 @@ Unless explicitly noted, integer encoding MUST be in big-endian format.
-`ExecutionAddress` is defined in [EIP-6466](./eip-6466.md).
-
-### Algorithmic Transaction (RLP)
-
-**This section is only applicable until [EIP-6404](./eip-6404.md) and [EIP-6466](./eip-6466.md) have been included within a hard fork.**
+### Algorithmic Transaction
 
+#### Format
 
 A new [EIP-2718](./eip-2718.md) transaction is defined with a structure of:
 
 ```python
 
 ALGORITHMIC_TX_TYPE || rlp([
-  algorithm_type,       # uint8
-  signature_info,       # Bytes
   tx_type,              # uint8
   tx_data,              # RLP encoded transaction data
   chain_id,             # uint64
   [
-    [
-      algorithm_type: uint8,
-      signature_info: Bytes
-    ],
+    additional_signature: Bytes
     ...
-  ]
+  ],
+  signature_info,       # Bytes
 ])
 
 ```
 
-The signing hash of the payload MUST be calculated with `keccak256(rlp([tx_type, tx_data, chain_id, additional_signatures]))`.
-
-If a signature is being overridden via an `additional_signatures` entry, the `r` field MUST be set to `keccak256(rlp([algorithm_type, signature_info]))` and the `v` and `y_parity` values MUST be set to `0x0`. This MUST be verified during transaction validation. Each entry MUST NOT be of type `secp256k1`.
-
-Each `additional_signatures` entry MUST be used at least once within the transaction and MUST not be duplicated. Non-used or duplicated entries will invalidate the tx. Entries MUST be sorted in ascending order (0x0 -> 0xff...ff) via `uint256(keccak256(rlp([algorithm_type, signature_info])))`.
+The first byte of `signature_info` (`signature_info[0]`) represents the algorithm used to sign the transaction in the `payload` field. All instances of `signature_info` MUST be at least 1 byte for the selector.
 
-The inner transaction MUST still contain the `r`, `s`, and `y_parity` which MUST be set to `0x0`.
+The `payload` transaction MUST still contain the `r`, `s`, and `y_parity` fields which MUST be set to `0x0`.
 
-It MUST also be checked that `len(additional_signatures)` <= `MAX_ADDITIONAL_INFO`
 
-### Algorithmic Transaction (SSZ)
+#### Signatures
 
-**This section is only applicable once [EIP-6404](./eip-6404.md) and [EIP-6466](./eip-6466.md) have been included within a hard fork.**
+The signing hash of the payload MUST be calculated with `keccak256(rlp([tx_type, tx_signing_hash, chain_id, additional_signatures]))`.
 
-`TransactionPayload` is defined as any type that derives its type from `Profile[TransactionPayload]`.
-
-This EIP introduces a new [EIP-6404](./eip-6404.md) transaction profile of SSZ type `AlgorithmicTransaction` defined below:
-
-```python
-
-class AdditionalSignature(ProgressiveContainer(active_fields=[1,1])):
-  # The selector for which algorithm is to be used.
-  algorithm_type: uint8,
-  
-  # The signature of the algorithm specified by `algorithm_type`
-  signature_info: ProgressiveList[byte],
+`tx_signing_hash` Is the hash specified by the respective transaction type denoted by the `tx_type` field.
```

Comment:
```suggestion
`tx_signing_hash` is the hash specified by the respective transaction type denoted by the `tx_type` field.
```

And, for EIP-155, it's also determined by the `v` value (have to fake 0-filled signature for signing hash)